Dockerfile

```diff
@@ -1 +1,41 @@
+ARG ARCH="amd64"
+ARG OS="linux"
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+LABEL maintainer="Armedia, LLC"
 
+ARG ARCH="amd64"
+ARG OS="linux"
+ARG VER="2.28.1"
+ARG PKG="prometheus"
+ARG SRC="${PKG}-${VER}.${OS}-${ARCH}"
+
+RUN curl \
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
+		-o package.tar.gz && \
+	tar -xzvf "package.tar.gz" && \
+	mkdir -pv "/prometheus" && \
+	mkdir -pv "/etc/prometheus" && \
+	mkdir -pv "/usr/share/prometheus" && \
+	mv -vif "${SRC}/LICENSE"                     "/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"                      "/NOTICE" && \
+	mv -vif "${SRC}/prometheus"                  "/bin/prometheus" && \
+	mv -vif "${SRC}/promtool"                    "/bin/promtool" && \
+	mv -vif "${SRC}/prometheus.yml"              "/etc/prometheus/prometheus.yml" && \
+	mv -vif "${SRC}/console_libraries/"          "/usr/share/prometheus/" && \
+	mv -vif "${SRC}/consoles/"                   "/usr/share/prometheus/" && \
+	ln -s "/usr/share/prometheus/console_libraries" "/etc/prometheus" && \
+	ln -s "/usr/share/prometheus/consoles"          "/etc/prometheus" && \
+	chown -R nobody:nobody "/etc/prometheus" "/prometheus" && \
+	rm -rvf "${SRC}" package.tar.gz
+
+#COPY npm_licenses.tar.bz2                        /npm_licenses.tar.bz2
+
+USER       nobody
+EXPOSE     9090
+VOLUME     [ "/prometheus" ]
```

Comment:
This is the first time I see this `VOLUME` command in a Dockerfile. Could you please explain to me what it does? Thanks

Reply:
The VOLUME command adds persistence beyond the container. I.e. the container's FS is created, and if a volume isn't attached to it as part of the container's startup, then one is created. This volume will contain persistent files separate from the container's so if the container goes down, the volume remains and can be re-used by a later container if desired.

The simpler way to use this is by defining named volumes which can easily be re-used (b/c their names are easily recognizable). This is how you can achieve data persistence for database containers, etc. Alfresco does this for its data store, for instance...

Reply:
All right, I see. So if Kubernetes already mounts a volume on `/prometheus`, Docker will not attempt to create one if its own volume, is that correct?

Reply:
Correct. You use "VOLUME" to indicate that the data contained within is to be persisted independently of the container's own persistence. I.e. you nuke the container, the volume stays so you can re-use it or recover its contents later. You can still do this by hard-mounting volumes unto directories, but it's generally a good idea to declare that your container expects a (set of) volume(s) to be provisioned for persistent storage.

Reply:
@drivera-armedia All right, that sounds good. I learned something new today!